pkg/registry/populator_test.go

```diff
@@ -1534,3 +1534,709 @@ func TestSemverPackageManifest(t *testing.T) {
+			db, cleanup := CreateTestDb(t)
+			defer cleanup()
+
+			load, err := sqlite.NewSQLLiteLoader(db, sqlite.WithEnableAlpha())
+			require.NoError(t, err)
+			err = load.Migrate(context.TODO())
+			require.NoError(t, err)
+			query := sqlite.NewSQLLiteQuerierFromDb(db)
+
+			graphLoader, err := sqlite.NewSQLGraphLoaderFromDB(db)
+			require.NoError(t, err)
+
+			populate := func(names []string) error {
+				refMap := make(map[image.Reference]string, 0)
+				for _, name := range names {
+					refMap[image.SimpleReference("quay.io/test/"+name)] = "../../bundles/" + name
+				}
+				return registry.NewDirectoryPopulator(
+					load,
+					graphLoader,
+					query,
+					refMap,
+					make(map[string]map[image.Reference]string, 0), false).Populate(registry.ReplacesMode)
+			}
+			// Initialize index with some bundles
+			require.NoError(t, populate(tt.args.bundles))
+
+			// Add the rebuilds one at a time
+			for _, rb := range tt.args.rebuilds {
+				require.NoError(t, populate([]string{rb}))
+			}
+
+			// Check graph is unchanged but has new csv name + version
+			// Ensure bundlePaths in db match
+			bundles, err := query.ListBundles(context.Background())
+			require.NoError(t, err)
+			var bundlePaths []string
+			for _, bundle := range bundles {
+				bundlePaths = append(bundlePaths, bundle.BundlePath)
+				bundleThatReplaces, _ := query.GetBundleThatReplaces(context.Background(), bundle.CsvName, bundle.PackageName, bundle.ChannelName)
+				if bundleThatReplaces != nil {
+					require.Equal(t, tt.expected.whatReplaces[bundle.CsvName][bundle.ChannelName], bundleThatReplaces.CsvName)
+				} else {
+					require.Equal(t, tt.expected.whatReplaces[bundle.CsvName][bundle.ChannelName], "")
+				}
+				substitution, err := getBundleSubstitution(context.Background(), db, bundle.CsvName)
+				require.NoError(t, err)
+				require.Equal(t, tt.expected.substitutions[bundle.CsvName], substitution)
+			}
+			require.ElementsMatch(t, tt.expected.bundles, bundlePaths)
+
+			// check default channel
+			defaultChannel, err := query.GetDefaultChannelForPackage(context.Background(), "prometheus")
+			require.NoError(t, err)
+			require.Equal(t, tt.expected.defaultChannel, defaultChannel)
+		})
+	}
+}
+
+func getBundleSubstitution(ctx context.Context, db *sql.DB, name string) (string, error) {
+	query := `SELECT substitutesfor FROM operatorbundle WHERE name=?`
+	rows, err := db.QueryContext(ctx, query, name)
+	if err != nil {
+		return "", err
+	}
+	defer rows.Close()
+
+	var substitutesFor sql.NullString
+	if rows.Next() {
+		if err := rows.Scan(&substitutesFor); err != nil {
+			return "", err
+		}
+	}
+	return substitutesFor.String, nil
+}
+
+func TestEnableAlpha(t *testing.T) {
+	type args struct {
+		bundles     []string
+		enableAlpha bool
+	}
+	type expected struct {
+		err error
+	}
+	tests := []struct {
+		description string
+		args        args
+		expected    expected
+	}{
+		{
+			description: "EnableAlpha/SubstitutesForTrue",
```

Comment:
nit: name stuttering, the test will be prefixed with EnableAlpha already

Reply:
ah right!